### PR TITLE
Issue 97

### LINKED ISSUE #97: Return all matches for a gene
Not just the strongest match per source

### PR DESCRIPTION
**Pull request by korikuzma**
Closes #97 

Notes: 
- Return all matches for a query. If different match types point to same record, use the highest match type
- Added match_type to Gene class
    - In ETL methods, temporarily added + removed this field to get Pydantic validation to pass

### CODE REVIEW

File: gene/etl/base.py [hunk 1]
```diff
@@ -94,7 +93,9 @@ def _load_gene(self, gene, batch) -> None:
         :param BatchWriter batch: Object to write data to DynamoDB
         """
         try:
+            gene['match_type'] = MatchType.NO_MATCH
             assert Gene(**gene)
```
Comment (jsstevenson) [inline gene/etl/base.py:97]: what about something like

```
assert Gene(match_type=MatchType.NO_MATCH, **gene)
```

File: gene/query.py [hunk 2]
```diff
@@ -160,79 +144,14 @@ def fill_no_matches(self, resp: Dict) -> Dict:
                     'records': [],
                     'source_meta_': self.fetch_meta(src_name)
                 }
+            else:
+                # sort records based on highest match_type
+                records = resp['source_matches'][src_name]['records']
+                if len(records) > 1:
+                    records = sorted(
+                        records, key=lambda k: k.match_type, reverse=True)
```
Comment (jsstevenson) [inline gene/query.py:152]: IMO this should dictate a name change for this method ("post process query" or something?) , or be moved further up in the call stack

**Review by jsstevenson — CHANGES_REQUESTED** [review]
👍 👍 👍

File: gene/etl/base.py (hunk 1 above)
Comment (korikuzma) [reply gene/etl/base.py:97]: ^ Tried that and getting TypeError, will just leave with adding/deleting the field

Comment (korikuzma) [reply gene/etl/base.py:97]: Oh wait I forgot I set this in NCBI, I'll try removing that and that should fix it

**Review by jsstevenson — DISMISSED** [review]

**Pull request merged**